### PR TITLE
don't enable live map for WIG mapping

### PR DESCRIPTION
**Pull request by ztNFny**
if map filter context is explicitly set to GeocacheFilterContext.FilterType.TRANSIENT don't enable Live mode, no matter what map type is displayed

@moving-bits there's a lot of "magic" happening in determining when to show which caches (only selected, stored, live) that's quite intransparent even after debugging this code. Thus I'm asking for your review as most of this seems to come from you.
Maybe this magic could also be replaced by a more transparent implementation, but I fear that'd be a big rewrite. Or maybe it's not so magic after all and I'm just missing something??

fixes https://github.com/cgeo/cgeo/issues/16338

### CODE REVIEW

**Comment by ztNFny** [comment]
marking as don't merge as this is probably not a good approach based on comments - hoping for a better implementation :-)

**Comment by ztNFny** [comment]
let's try this in another way. As UMTT_Viewport is not used outside of WIG it shouldn't have any additional impact

**Review by moving-bits — CHANGES_REQUESTED** [review]
> let's try this in another way. As UMTT_Viewport is not used outside of WIG it shouldn't have any additional impact

This seems to work, technically, for displaying the zone without any other cache, similar to navigating to a single cache, but still gives an inconsistent UX:
- on mapping a single cache with "navigate to", the map's title changes to the name of this cache,
- whereas on "show this zone on map", the map title stays at "offline", though not displaying any offline caches.

Thus, if pursuing this way, we should use the zone name as title (see `UnifiedMapActivity.calculateTitle()`), and we should rename the `UnifiedMapTypeType` from `UMTT_Viewport` to something WIG-related.

**Comment by ztNFny** [comment]
> Thus, if pursuing this way, we should use the zone name as title (see `UnifiedMapActivity.calculateTitle()`), and we should rename the `UnifiedMapTypeType` from `UMTT_Viewport` to something WIG-related.

While that Feature Request is unrelated to this PR I nevertheless added it to get this PR passed.

File: main/src/main/java/cgeo/geocaching/maps/DefaultMap.java [hunk 1]
```diff
@@ -146,4 +146,12 @@ public static void startActivityViewport(final Activity fromActivity, final View
             mapType.launchMap(fromActivity);
         }
     }
+
+    public static void startActivityWherigoMap(final Activity fromActivity, final Viewport viewport, final String mapTitle) {
+        if (Settings.useUnifiedMap()) { // only supported for UnifiedMap
+            Log.e("Launching UnifiedMap in viewport mode, viewport=" + viewport + ")");
```
Comment (eddiemuc) [inline main/src/main/java/cgeo/geocaching/maps/DefaultMap.java:152]: Use `Log.iForce()` for info logs which should appear in the logfile anyway.

File: main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapType.java [hunk 2]
```diff
@@ -55,6 +55,13 @@ public UnifiedMapType(final Viewport viewport) {
         this.viewport = viewport;
     }
 
+    /** open map and scale to a given viewport */
+    public UnifiedMapType(final Viewport viewport, final String title) {
```
Comment (eddiemuc) [inline main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapType.java:59]: With this new constructor in place, I assume that the constructor taking only a Viewport (not a name) is no longer used. Thus it should be removed.

**Review by eddiemuc — APPROVED** [review]
LGTM. I have two minor comments only

File: main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapType.java (hunk 2 above)
Comment (eddiemuc) [reply main/src/main/java/cgeo/geocaching/unifiedmap/UnifiedMapType.java:59]: I introduced it for Wherigos... it was not there before. But as said, it is minor. No harm in leaving it in

**Pull request merged**